src/app/core/tools/handlers/SelectPathHandler.ts

```diff
@@ -27,8 +34,17 @@ export const SelectPathHandler: EventHandler = ({
             .getWires()
             .reverse()
             .find(o => o.isWithinSelectBounds(worldMousePos));
-        const path = GetPath(wire!);
+        const component = designer
+            .getObjects()
+            .reverse()
+            .find(o => o.isWithinSelectBounds(worldMousePos));
```

Comment:
Combine these using `getAll()`, which, when reversed, will have the wires in front and the objects at the back which is the desired behavior. (also make sure to cast `as (Component | Wire)`)

Then for the if-statement below, use `(obj instanceof Wire)`